airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_job_repository.py

```diff
@@ -0,0 +1,167 @@
+            raise AirbyteTracedException(
+                internal_message="Polling Requester received an empty Response.",
+                failure_type=FailureType.system_error,
+            )
+        return polling_response
+
+    def _get_validated_job_status(self, response: requests.Response) -> AsyncJobStatus:
+        """
+        Validates the job status extracted from the API response.
+
+        Args:
+            response (requests.Response): The API response.
+
+        Returns:
+            AsyncJobStatus: The validated job status.
+
+        Raises:
+            ValueError: If the API status is unknown.
+        """
+
+        api_status = next(iter(self.status_extractor.extract_records(response)), None)
+        job_status = self.status_mapping.get(str(api_status), None)
+        if job_status is None:
+            raise ValueError(
+                f"API status `{api_status}` is unknown. Contact the connector developer to make sure this status is supported."
+            )
+
+        return job_status
+
+    def _start_job_and_validate_response(self, stream_slice: StreamSlice) -> requests.Response:
+        """
+        Starts a job and validates the response.
+
+        Args:
+            stream_slice (StreamSlice): The stream slice to be used for the job.
+
+        Returns:
+            requests.Response: The response from the job creation requester.
+
+        Raises:
+            AirbyteTracedException: If no response is received from the creation requester.
+        """
+
+        response: Optional[requests.Response] = self.creation_requester.send_request(stream_slice=stream_slice)
+        if not response:
+            raise AirbyteTracedException(
+                internal_message="Always expect a response or an exception from creation_requester",
+                failure_type=FailureType.system_error,
+            )
+
+        return response
+
+    def start(self, stream_slice: StreamSlice) -> AsyncJob:
+        """
+        Starts a job for the given stream slice.
+
+        Args:
+            stream_slice (StreamSlice): The stream slice to start the job for.
+
+        Returns:
+            AsyncJob: The asynchronous job object representing the started job.
+        """
+
+        response: requests.Response = self._start_job_and_validate_response(stream_slice)
+        job_id: str = str(uuid.uuid4())
+        self._create_job_response_by_id[job_id] = response
+
+        return AsyncJob(api_job_id=job_id, job_parameters=stream_slice)
+
+    def update_jobs_status(self, jobs: Iterable[AsyncJob]) -> None:
+        """
+        Updates the status of multiple jobs.
+
+        Args:
+            jobs (Iterable[AsyncJob]): An iterable of AsyncJob objects representing the jobs to update.
+
+        Returns:
+            None
+        """
+        for job in jobs:
+            stream_slice = StreamSlice(
+                partition={"create_job_response": self._create_job_response_by_id[job.api_job_id()]},
```

Comment:
Because we don't have interpolation on random fields, we have this hack which consist on using the stream_slice to allow for interpolation. This is what it looks like in the manifest: https://github.com/airbytehq/airbyte/compare/async-job/feature-branch#diff-b041c21db61fd5fe7fb67481e6167b6a343236571089c1ed61498b597202a750R913

Reply:
maybe worth leaving this comment in the code?